Source/DafnyCore/AST/Statements/Assignment/AssignOrReturnStmt.cs

```diff
@@ -23,7 +23,7 @@ public class AssignOrReturnStmt : ConcreteUpdateStatement, ICloneable<AssignOrRe
     }
   }
 
-  public override IEnumerable<INode> Children => ResolvedStatements;
+  public override IEnumerable<INode> Children => ResolvedStatements.Concat(Proof != null ? Proof.Children : new List<INode>());
```

Comment:
Small thing but `Enumerable.Empty<INode>()` will do the same thing as `new List<INode>()` here but save you the allocation, and `Proof != null ? Proof.Children : <alt>` could be `Proof?.Children ?? <alt>`

Reply:
Good idea!